cdap-explore-client/src/main/java/co/cask/cdap/explore/client/AbstractExploreClient.java

```diff
@@ -316,17 +319,23 @@ public void run() {
                 onSuccess(handle);
               }
             }, 300, TimeUnit.MILLISECONDS);
-          } else {
-            if (!status.hasResults()) {
-              close(handle);
-            }
-            if (!resultFuture.set(new ClientExploreExecutionResult(AbstractExploreClient.this,
-                                                                   handle, status))) {
-              close(handle);
-            }
+            return;
+          }
+          if (QueryStatus.OpStatus.ERROR.equals(status.getStatus())) {
+            throw new SQLException(status.getErrorMessage(), status.getSqlState());
```

Comment:
No, it gets caught by the catch in line 329, which will set it as the exception on the future. I could have done that here and returned, but then also mustCloseHandle has to be set etc. I am just throwing it here because I wanted to have a single outer catch catch and close to deal with all failures.